src/field.rs

```diff
@@ -219,7 +219,7 @@ pub(crate) trait FieldElementExt: FieldElement {
         let one = Self::Integer::from(Self::one());
         let mut encoded = Vec::with_capacity(bits);
         for _ in 0..bits {
-            let w = Self::from(i & one);
+            let w = Self::try_from(i & one).unwrap();
```

Comment:
`try_from` should return `Result<_, FieldError>` here so I think we can propagate the error with `?`
```suggestion
            let w = Self::try_from(i & one)?;
```